tests/foreman/ui_airgun/test_contenthost.py

```diff
@@ -14,103 +14,502 @@
-        and errata installation are successful
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
 
-    :CaseLevel: System
-    """
-    org = entities.Organization().create()
-    lce = entities.LifecycleEnvironment(organization=org).create()
+
+@fixture(scope='module', autouse=True)
+def repos_collection(module_org):
+    """Adds required repositories, AK, LCE and CV for content host testing"""
+    lce = entities.LifecycleEnvironment(organization=module_org).create()
     repos_collection = RepositoryCollection(
         distro=DISTRO_RHEL7,
         repositories=[
+            RHELAnsibleEngineRepository(cdn=True),
             SatelliteToolsRepository(),
-            YumRepository(url=FAKE_1_YUM_REPO)
+            YumRepository(url=FAKE_1_YUM_REPO),
+            YumRepository(url=FAKE_6_YUM_REPO)
         ]
     )
-    repos_collection.setup_content(org.id, lce.id, upload_manifest=True)
+    repos_collection.setup_content(module_org.id, lce.id, upload_manifest=True)
+    return repos_collection
+
+
+@fixture
+def vm(repos_collection):
+    """Virtual machine registered in satellite with katello-agent installed"""
     with VirtualMachine(distro=DISTRO_RHEL7) as vm:
         repos_collection.setup_virtual_machine(vm)
-        result = vm.run('yum -y install {0}'.format(FAKE_1_CUSTOM_PACKAGE))
-        assert result.return_code == 0
-        with session:
-            session.organization.select(org.name)
-            # Ensure content host is searchable
-            assert session.contenthost.search(
-                vm.hostname)[0]['Name'] == vm.hostname
-            chost = session.contenthost.read(vm.hostname)
-            # Ensure all content host fields/tabs have appropriate values
-            assert chost['details']['name'] == vm.hostname
-            assert (
-                chost['details']['content_view'] ==
-                repos_collection.setup_content_data['content_view']['name']
-            )
-            assert chost['details']['lce'][lce.name][lce.name]
-            assert (
-                chost['details']['registered_by'] == 'Activation Key {}'
-                .format(repos_collection.setup_content_data[
-                    'activation_key']['name'])
-            )
-            assert chost['provisioning_details']['name'] == vm.hostname
-            assert (
-                chost['subscriptions']['resources']['assigned'][0][
-                    'Repository Name'] ==
-                repos_collection.custom_product['name']
+        yield vm
+
+
+def set_ignore_facts_for_os(value=False):
+    """Helper to set 'ignore_facts_for_operatingsystem' setting"""
+    ignore_setting = entities.Setting().search(
+        query={'search': 'name="ignore_facts_for_operatingsystem"'})[0]
+    ignore_setting.value = str(value)
+    ignore_setting.update({'value'})
+
+
+@tier3
+def test_positive_end_to_end(session, repos_collection, vm):
```

Comment:
just curious: repos_collection fixture is set autouse=True , are we able to omit it's declaration?

Reply:
we can't, if we don't pass it as arg, the test won't receive fixture's output, and attempt to access `repos_collection` will give us function - as this module has a function with such name :)